src/main/java/net/minecraftforge/fml/hooks/BasicEventHooks.java

```diff
@@ -119,4 +121,16 @@ public static void onPostServerTick()
     {
         MinecraftForge.EVENT_BUS.post(new TickEvent.ServerTickEvent(TickEvent.Phase.END));
     }
+    
+    public static void onPreBlockEntityTick(TileEntity blockEntity)
+    {
+        MinecraftForge.EVENT_BUS.post(new TickEvent.BlockEntityTickEvent(TickEvent.Phase.START, blockEntity));
+    }
+
+    public static void onPostBlockEntityTick(TileEntity blockEntity)
+    {
+        if (blockEntity.getLevel() != null)
+            ForgeNetwork.sendBlockEntityCapabilities(blockEntity, false);
+        MinecraftForge.EVENT_BUS.post(new TickEvent.BlockEntityTickEvent(TickEvent.Phase.END, blockEntity));
+    }
```

Comment:
So this is can be the main concern this pr bring, is this event a good or a bad thing it should be discussed.
Maybe it could be better to comment out the event and create a separate pr for it .

Reply:
Erm yes, I am getting performance anxiety by just reading this

Reply:
I'm not a big fan of it either but currently some mods found a way to do it and it's ugly and performance hungry.

Reply:
Well it's up to them to make sure they sync lazily, Forge probably shouldn't introduce such broken behavior.

Reply:
> I'm not a big fan of it either but currently some mods found a way to do it and it's ugly and performance hungry.

I was talking about the event some mods managed to do TE ticking events from world ticking event.

Reply:
I think I'm gonna scrap the event; if somebody needs it a new PR can be opened. 

Also, in terms of the performance of Forge syncing caps, only caps that ask to be actively synced will be synced and if no caps request a sync then no packet will be sent. Moreover, only dirty data will serialised - I've tried to make it as slimline as possible. A flag is passed into the encode method of the cap to distinguish between syncing (only if requested) or the parent object requiring serialisation (such as if an `ItemStack` is written to a `PacketBuffer`, `Entity` starts getting tracked by a player etc.

Reply:
Yeah i was only really concerned about the two event dispatches in the tight loop. We've been burned by the dispatch in the ItemStack ctor and this just gave me PTSD :)